armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -73,7 +98,13 @@ def setAssembly(self, a, setFuel=True):
         ----------
          a : :py:class:`Assembly <armi.reactor.assemblies.Assembly>` object.
             ARMI assembly to be changed
+        setFuel : boolean, optional
+            Boolean to determine whether or not fuel blocks should have their target components set
+            This is useful when target components within a fuel block need to be determined on-the-fly.
         """
+        self._detailedAxialExpansion = Settings.instance.__getitem__(
```

Comment:
This won't work because you need to pass in a cs object into the method to get the actual cases settings to use. Otherwise this will also resolve to the default I believe. We want to move away from having global case settings.

Reply:
I looked over this more closely and it does work because we have a global CS available in ARMI, but I believe that the design goal for ARMI is to remove the master CS. Is this still true @ntouran? Nonetheless I think that after we spoke offline it's better to just pass in `detailedAxialExpansion` as a kwarg on the init for this changer.

Reply:
We _do_ want to stop using `getMasterCS()`, as it is somewhat problematic. I would _love_ to remove the global CS state, but it will not happen today.

Reply:
addressed. 

https://github.com/terrapower/armi/commit/286c294a6bb6f0f91f5b0ea7c230912f62fbcd29